modules/calc/src/main/java/com/opengamma/strata/calc/runner/function/CurrencyConvertible.java

```diff
@@ -17,11 +17,30 @@
 public interface CurrencyConvertible<R> {
 
   /**
-   * Returns a copy of the object with any currency amounts converted into the reporting currency.
+   * Checks whether this object can be currency converted.
+   * <p>
+   * This will return true if some form of currency conversion will be attempted
+   * by {@link #convertedTo(Currency, CalculationMarketData)}.
+   * 
+   * @return true if currency conversion is possible
+   */
+  public abstract boolean isConvertible();
```

Comment:
Why is this method required? The design we discussed was for `convertedTo` to simply return `this` when called on something that isn't actually convertible.

It is also a blatant violation of tell-don't-ask.